wagtail/contrib/redirects/models.py

```diff
@@ -204,6 +209,10 @@ def normalise_page_route_path(url):
 
         return path
 
+    @staticmethod
+    def get_redirect_hash(url):
+        return uuid.uuid5(REDIRECT_NAMESPACE_UUID, url)
```

Comment:
I recently learned that `uuid3` does the same thing as `uuid5`, but uses MD5 for hashing instead of SHA-1, which tends to be faster/less demanding. Since we gain nothing from the additional security features of SHA-1, it's probably a worthwhile swap in this context.
```suggestion
        return uuid.uuid3(REDIRECT_NAMESPACE_UUID, url)
```

Reply:
Actually, scratch this! Python's `hashlib` module has included `blake2b()` and `blake2s()` for  long enough now that it's even available in Wagtail's advertised minumum compatible Python version (3.9). So, we should find a way to use that if we can. I know I originally suggested the `UUIDField` approach, but there's no real reason to stick to the UUID format if there are better hashing options available to us.

Reply:
Heck, I might even suggest adding https://github.com/BLAKE3-team/BLAKE3 as a dependency and going with that.

Reply:
This might all seem a little fussy for a small internal model, but redirect lookups are essentially user-facing, as they initiated for pretty much every request that would otherwise raise a 404 (which can be a tonne of requests - especially when launching a new site or making significant changes to information architecture). It's more important here to make those hash conversions efficient than it is for some other places (e.g. the reference index, which is internal only)

Reply:
Intuitively it seems to me that the database lookup (over an unbounded number of records) is going to be the bottleneck rather than computing 2-4 hashes, so unless there's solid evidence to suggest otherwise, I'd go for `uuid3` and a `UUIDField`.